### ISSUE TITLE
avm2: All elements in an Array should be enumerable

### ISSUE THREAD

**Issue by EmperorBale** [ruffle-rs/ruffle#5162]
The current code does not work in Ruffle:
```as
var a = [1, 2, 3, 4, 5];
for each(var i in a) {
    trace(i)
}
```
Because the elements in the Array are not enumerable. Array objects should have a custom `get_enumerant_name` and `property_is_enumerable` implementation to support this.

**Comment by danielhjacobs** [comment]
This was fixed #5447, correct?

**Comment by EmperorBale** [comment]
> This was fixed #5447, correct?

Yup
